src/main/java/edu/harvard/iq/dataverse/DatasetServiceBean.java

```diff
@@ -553,28 +547,18 @@ public DatasetLock addDatasetLock(Long datasetId, DatasetLock.Reason reason, Lon
     }
 
     @TransactionAttribute(TransactionAttributeType.REQUIRES_NEW)
-    public void removeDatasetLock(Long datasetId) {
+    public void removeDatasetLock(Long datasetId, DatasetLock.Reason aReason) {
```

Comment:
It looks like with this implementation that multiple locks could be removed if they share the same reason. I am assuming that we are ensuring only one lock per reason is added, or that in all cases we want all locks of a reason to be removed. Even if one of these two are true, maybe this should be renamed to reflect what is happening with the filter.